core-api/src/main/java/org/silverpeas/core/wopi/WopiFileEditionManager.java

```diff
@@ -0,0 +1,153 @@
+ * "https://www.silverpeas.org/legal/floss_exception.html"
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.silverpeas.core.wopi;
+
+import org.silverpeas.core.util.ServiceProvider;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.BiFunction;
+
+/**
+ * In charge of the management of WOPI contexts.
+ * <p>
+ * The exposed signatures permits to caller to initialize a WOPI edition.
+ * </p>
+ * <p>
+ * WOPI services can use directly the implementation of this interface in order to get more
+ * functionality.
+ * </p>
+ * @author silveryocha
+ */
+public interface WopiFileEditionManager {
+
+  static WopiFileEditionManager get() {
+    return ServiceProvider.getService(WopiFileEditionManager.class);
+  }
+
+  /**
+   * Notifies the manager of the number of users which are editing the given file at an instant.
+   * @param file a {@link WopiFile} a file.
+   * @param userIds set of wopi user identifiers.
+   */
+  void notifyEditionWith(final WopiFile file, final Set<String> userIds);
+
+  /**
+   * Gets the list of file edited by the given user.
+   * @param user a {@link WopiUser} instance.
+   * @return a list of {@link WopiFile}.
+   */
+  List<WopiFile> getEditedFilesBy(final WopiUser user);
+
+  /**
+   * Gets the list of users which are editor of the given file.
+   * @param file a {@link WopiFile} instance.
+   * @return a list of {@link WopiUser}.
+   */
+  List<WopiUser> getEditorsOfFile(final WopiFile file);
+
+  /**
+   * Enables the services.
+   * <p>
+   * This usable only in the case of WOPI is enable by setting property file.<br/>
+   * When enabled by setting file, it is possible the enable/disable the services in order to get
+   * as fast as possible control on WOPI exchanges.
+   * </p>
+   * @param enable true to enable, false to disable.
+   */
+  void enable(final boolean enable);
+
+  /**
+   * Indicates if the WOPI edition is enabled.
+   * @return true if enabled, false otherwise.
+   */
+  boolean isEnabled();
+
+  /**
+   * Indicates if given {@link WopiFile} is handled by WOPI client.
+   * @param file the wopi file to check.
+   * @return true if handled, false otherwise.
+   */
+  boolean isHandled(WopiFile file);
+
+  /**
+   * Initializing a WOPI edition from given data.
+   * @param request the current request from which the edition is started.
+   * @param file a Silverpeas's WOPI file.
+   * @return an optional URL of the Silverpeas's editor page.
+   */
+  Optional<String> initializeWith(HttpServletRequest request, WopiFile file);
```

Comment:
What's the fuck?! An HttpServletRequest ref here in core-api!!! We have already to deal with old spaghetti code in which web parts are interleaved with api (or api implem) parts, so please don't do it more worse.
Please, the core-api must not ref any code and concepts in the Java Web Profile! core-api is about business model (and utility codes) and must not have any knowledge upon the way it is used! Web code should be built upon core-api, not the reverse.
If this manager has to handle anything coming from the Web, then his place should be in core-web. If this interface has to be used in core-api or core-lib, then split this interface in two by moving any web-related code from this interface into another interface defined itself in core-web.